bin/check-policies.sh

```diff
@@ -34,20 +37,21 @@ check_policies() {
   done
   yq r -j $constraints_file | jq '{parameters: .}' | yq r -P - >$parameters_file
 
-  # validate_resources
   echo "Checking manifests against policies for $cluster_env cluster."
   conftest test --fail-on-warn --all-namespaces -d "$parameters_file" -p $policies_path "$k8s_resources_path/$k8s_version" || exitcode=1
-  [ "$CI" = 'true' ] && [ $exitcode -ne 0 ] && exit $exitcode
+  [ -n "$CI" ] && [ $exitcode -ne 0 ] && exit $exitcode
   return 0
 }
 
 ! $(yq r $otomi_settings "otomi.addons.conftest.enabled") && echo "skipping" && exit 0
 
-if [ "${1-}" != "" ]; then
-  echo "Checking policies for cluster '$(cluster_env)'"
-  shift
-  check_policies "$@"
-else
-  echo "Checking policies for all clusters"
-  for_each_cluster check_policies
+function main() {
+  validate_resources check_policies "$@"
```

Comment:
This is the main refactor of `check_policies`, `validate-templates` and `validate-values`. I am actually quite satisfied with the solution. They all have the same pattern actually so this removes a lot of duplication and creates opportunity for re-use.